src/bacnet/basic/bbmd6/h_bbmd6.c

```diff
@@ -612,6 +612,46 @@ static void bbmd6_address_resolution_handler(
     }
 }
 
+/**
+ * Handler for Forwarded-Address-Resolution
+ *
+ * @param addr - BACnet/IPv6 source address any NAK or reply back to.
+ * @param pdu - The received NPDU+APDU buffer.
+ * @param pdu_len - How many bytes in NPDU+APDU buffer.
+ */
+static void bbmd6_forwarded_address_resolution_handler(
+    const BACNET_IP6_ADDRESS *addr, const uint8_t *pdu, uint16_t pdu_len)
+{
+    int function_len = 0;
+    uint32_t vmac_src = 0;
+    uint32_t vmac_target = 0;
+    uint32_t vmac_me = 0;
+    BACNET_IP6_ADDRESS bip6_address = {
+        0,
+    };
+
+    if (addr && pdu) {
```

Comment:
I think logic would read better if you specified the addr and pdu values explicitly, like `addr != NULL && pdu != 0`